src/modules/users/users.service.ts

```diff
@@ -10,6 +16,22 @@ export class UsersService {
     private usersRepository: Repository<User>,
   ) {}
 
+  async createUser(createUserDto: CreateUserDto): Promise<User> {
+    const errors = await validate(createUserDto);
+    if (errors.length > 0) {
+      throw new BadRequestException('Validation failed');
+    }
+    const existingUser = await this.usersRepository.findOne({
+      where: { email: createUserDto.email },
+    });
+    if (existingUser) {
+      throw new ConflictException('Email already exists');
```

Comment:
Create una carpetita mas en el src que tenga todas las constantes que se van a utilizar en la app.
-> src/shared/constants.ts  
La idea es que todos los textos como: 'Email already exists' se encuentren dentro de una constate. De esta manera si mas adelante se quiere cambiar el mensaje lo cambiamos desde el archivo de constantes y no estan buscando dentro de la app. Te dejo un ejemplo de lo que tendría ese archivo: export const HTTP_STATUS_MESSAGES = {
  USER_DELETED_SUCCESS: 'User deleted successfully',
  USER_NOT_FOUND: 'User not found',
  FAILED_DELETED_USER: 'Failed to delete user',
  FAILED_FETCHING_USERS: 'Error occurred while fetching users',
  FAILED_FETCHING_GROUPS: 'Error occurred while fetching groups',
  GROUP_NOT_FOUND: 'Group not found',
  FAILED_DELETED_GROUP: 'Failed to delete group',
  GROUP_DELETED_SUCCESS: 'Group deleted successfully',
  ERROR_WHILE_CREATING_USER: 'Error occurred while creating user:',
  FAILED_DELETED_PLAYER: 'Failed to delete player',
  PLAYER_DELETED_SUCCESS: 'Player deleted successfully',
  PLAYER_NOT_FOUND: 'Player not found',
};  Despues en tu servicio solo importas el el mensaje desde tu objeto de constantes

Reply:
Si señor 🫡 Gracias señor 🫡, fuera de joda es muy util jajaja xD gracias